.azure-pipelines-templates/deploy_aci.yml

```diff
@@ -60,13 +93,46 @@ jobs:
     displayName: "Cleanup ACI"
     pool:
       vmImage: ubuntu-20.04
-    dependsOn: ${{ parameters.used_by }}
+    dependsOn:
+      - deploy_aci
+      - ${{ parameters.used_by }}
     condition: always()
+    variables:
+      resource_group: $[ dependencies.deploy_aci.outputs['deploy_aci.resource_group'] ]
     steps:
       - template: azure_cli.yml
+        parameters:
+          app_id: $(CCF_SNP_CI_APP_ID)
+          service_principal_password: $(CCF_SNP_CI_SERVICE_PRINCIPAL_PASSWORD)
+          tenant: $(CCF_SNP_CI_TENANT)
 
       - script: |
           set -ex
-          python3.8 scripts/azure_deployment/arm_template.py remove aci --subscription-id $(CCF_AZURE_SUBSCRIPTION_ID) --resource-group ccf-aci --aci-type dynamic-agent --deployment-name ci-$(Build.BuildNumber)
+          python3.8 scripts/azure_deployment/arm_template.py remove aci --subscription-id $(CCF_AZURE_SUBSCRIPTION_ID) --resource-group $resource_group --aci-type dynamic-agent --deployment-name ci-$(Build.BuildNumber)
         name: cleanup_aci
         displayName: "Delete the ACI and Azure Deployment"
+
+  - ${{ if eq(parameters.deploy_attestation_container, true) }}:
+      - job: cleanup_attestation_container_image
+        displayName: "Cleanup Attestation Container image"
+        pool:
+          vmImage: ubuntu-20.04
+        dependsOn:
+          - deploy_aci
+          - ${{ parameters.used_by }}
+        variables:
+          image_version: $[ dependencies.deploy_aci.outputs['build_and_push_attestation_container.ver'] ]
+        condition: always()
+        steps:
+          - template: azure_cli.yml
+            parameters:
+              app_id: ${{ parameters.app_id }}
+              service_principal_password: ${{ parameters.service_principal_password }}
+              tenant: ${{ parameters.tenant }}
+
+          - script: |
+              set -ex
+              az acr repository delete --name attestationcontainerregistry --image attestation-container:$image_version --yes
+
+            name: cleanup_attestation_container_image
+            displayName: "Delete the container image"
```

Comment:
If we delete the images after each run, if we run the pipeline periodically we might rebuild an image that could have just been re-used, we could have a cleanup task similar to what we do for the SNP pipeline?

Reply:
It should effective. I'll check what it does

Reply:
@DomAyre  I couldn't find the code to clean up docker images (not ACI containers) in the SNP pipeline. Can you tell me where it is please?

Reply:
Memo: It's done with [ACR task](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-tasks-overview) using azure cli (without any task definition file currently)

Reply:
I'll work on it in https://github.com/microsoft/CCF/issues/4844